src/Parsers/ASTUndropQuery.cpp

```diff
@@ -0,0 +1,38 @@
+#include <Parsers/ASTUndropQuery.h>
+#include <Common/quoteString.h>
+#include <IO/Operators.h>
+
+
+namespace DB
+{
+
+String ASTUndropQuery::getID(char delim) const
+{
+    return "UndropQuery" + (delim + getDatabase()) + delim + getTable();
+}
+
+ASTPtr ASTUndropQuery::clone() const
+{
+    auto res = std::make_shared<ASTUndropQuery>(*this);
+    cloneOutputOptions(*res);
+    cloneTableOptions(*res);
+    return res;
+}
+
+void ASTUndropQuery::formatQueryImpl(const FormatSettings & settings, FormatState &, FormatStateStacked) const
+{
+    settings.ostr << (settings.hilite ? hilite_keyword : "");
+    settings.ostr << "UNDROP ";
+    settings.ostr << "TABLE ";
+    settings.ostr << (settings.hilite ? hilite_none : "");
+
+    assert (table);
+    if (!database)
+        settings.ostr << backQuoteIfNeed(getTable());
+    else
+        settings.ostr << backQuoteIfNeed(getDatabase()) + "." << backQuoteIfNeed(getTable());
+
+    formatOnCluster(settings);
```

Comment:
UUID must be formatted as well (please add a test with `UNDROP ... UUID ... ON CLUSTER`)